### PR TITLE
refactor: separation between examples and integration tests renaming

### PR DESCRIPTION
**Pull request by aureamunoz**
Related to issue #256
Here's what I did (I hope this helps)

- I've renamed integration tests within `component-annotation/it` in order to have more details about what it checks in the names.
- The old `/examples/component-example` actually was, in my opinion, an integration test for the overriding of annotation config by values setted in a `application.yml` file so I moved this to `component-annotations/src/it/component-feat-229-override-annotationbased-config`
- The new `/examples/component-example` contains the simplest use case for `@ComponentAppliciation` without any specifique config so it checks the default values.

### CODE REVIEW

File: examples/component-example/src/main/java/io/dekorate/examples/component/Main.java [hunk 1]
```diff
@@ -22,13 +22,10 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
-@ComponentApplication(name = "hello-spring-boot", exposeService = true, envs = @Env(name = "key1", value = "val1"))
-@Link(name = "should-be-overridden-by-properties", componentName = "should-be-overridden-by-properties")
-@SpringBootApplication
```
Comment (iocanel) [inline examples/component-example/src/main/java/io/dekorate/examples/component/Main.java]: I would keep spring, as its a nice way of show casing how the `runtime`, `version` properties of component are popullated.

The rest I agree that should go.

File: annotations/component-annotations/src/it/component-feat-275-provide-port/pom.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
 		<relativePath/> <!-- lookup parent from repository -->
 	</parent>
 	<groupId>io.dekorate</groupId>
-	<artifactId>issue-275</artifactId>
+	<artifactId>component-feat-275-provide-port</artifactId>
```
Comment (iocanel) [inline annotations/component-annotations/src/it/component-feat-275-provide-port/pom.xml]: Don't have versy strong feelings about it, but I do feel that `component-` is a bit redundant, given that the tests live under `component-annoations`.

File: examples/component-example/src/main/java/io/dekorate/examples/component/Main.java (hunk 1 above)
Comment (aureamunoz) [reply examples/component-example/src/main/java/io/dekorate/examples/component/Main.java]: Ok, I agree, I hadn't thought about it. I'm gonna keep spring :-)

File: annotations/component-annotations/src/it/component-feat-275-provide-port/pom.xml (hunk 2 above)
Comment (aureamunoz) [reply annotations/component-annotations/src/it/component-feat-275-provide-port/pom.xml]: Yes, good remark. I didn't like it completely but I didn't know why.

**Pull request merged**